### PR TITLE
Add lexical validation for variable not present

### PR DESCRIPTION
**Pull request by ShubhamUniyal**

### CODE REVIEW

File: test/syntax.js [hunk 1]
```diff
@@ -74,6 +75,20 @@ describe('expression', function () {
...
     })
 
+    it('should validate simple expression', function () {
+      expect(validateExp('1<2', scope)).to.deep.equal([]);
+      expect(validateExp('1<!2', scope)).to.deep.equal(["cannot eval '!2' as value"]);
+      expect(validateExp('y<2', scope)).to.deep.equal(["y variable not present"]);
+      expect(validateExp('y<!2', scope)).to.deep.equal(["y variable not present", "cannot eval '!2' as value"]);
+    })
+
+    it('should validate complex expression', function () {
```
Comment (codebreach) [inline test/syntax.js]: what about checking that y is present and all..?

File: src/syntax.js [hunk 2]
```diff
@@ -51,6 +51,41 @@ function isFalsy (val) {
...
 }
 
+function validateExpression(exp, scope, errors = []) {
+  assert(scope, 'unable to evalExp: scope undefined')
+  var operatorREs = lexical.operators
+  var match;
+  for (var i = 0; i < operatorREs.length; i++) {
+    var operatorRE = operatorREs[i]
+    var expRE = new RegExp(`^(${lexical.quoteBalanced.source})(${operatorRE.source})(${lexical.quoteBalanced.source})$`)
+    if ((match = exp.match(expRE))) {
```
Comment (codebreach) [inline src/syntax.js]: can u autoformat this new code only?

like theres double `((` here for no reason

File: src/syntax.js [hunk 3]
```diff
@@ -51,6 +51,41 @@ function isFalsy (val) {
   return val === false || undefined === val || val === null
 }
 
+function validateExpression(exp, scope, errors = []) {
+  assert(scope, 'unable to evalExp: scope undefined')
+  var operatorREs = lexical.operators
+  var match;
+  for (var i = 0; i < operatorREs.length; i++) {
+    var operatorRE = operatorREs[i]
+    var expRE = new RegExp(`^(${lexical.quoteBalanced.source})(${operatorRE.source})(${lexical.quoteBalanced.source})$`)
```
Comment (codebreach) [inline src/syntax.js]: cant use template strings - this does not go through a transpiler

File: src/syntax.js [hunk 4]
```diff
@@ -51,6 +51,41 @@ function isFalsy (val) {
...
+  if (!str) return `Invalid Operator Usage`;
+
+  if (lexical.isLiteral(str)) {
+    return;
+  }
+  if (lexical.isVariable(str)) {
+    if(scope.get(str) !== undefined && scope.get(str) !== null) {
+      return;
+    } else {
+      return `${str} variable not present`
```
Comment (codebreach) [inline src/syntax.js]: don't use template strings?

i am not sure if they are safe. if we were using them before then you can keep them

Comment (ShubhamUniyal) [reply src/syntax.js]: yes, we were using template strings in this module before.

Comment (codebreach) [reply src/syntax.js]: OK LGTM format if possible

**Pull request merged**